.github/workflows/test_ci.yml

```diff
@@ -0,0 +1,31 @@
+# This workflow will install Python dependencies and run tests with a single version of Python for Penn repository
+
+name: Testing Penn
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```

Comment:
I'd probably add a matrix here to prepare running the tests on multiple versions of python and I'd check if the tests pass on macOS and Windows as well.